packages/shared/src/types/operations.ts

```diff
@@ -0,0 +1,37 @@
+import { ComponentOptions } from 'vue';
+import { Accountability } from './accountability';
+import { ApiExtensionContext } from './extensions';
+import { Field } from './fields';
+import { DeepPartial } from './misc';
+import { FlowRaw } from './flows';
+
+type OperationContext = ApiExtensionContext & {
+	data: Record<string, unknown>;
+	accountability: Accountability | null;
+};
+
+export type OperationHandler<Options = Record<string, unknown>> = (
+	options: Options,
+	context: OperationContext
+) => unknown | Promise<unknown> | void;
+
+export interface OperationAppConfig {
+	id: string;
+	name: string;
+	icon: string;
+	description?: string;
+	preview:
```

Comment:
Interfaces have an option named `preview` that points to the "preview thumbnail" image that is shown when configuring a field using the simplified field creation flow. In case we ever want to add something like this to operations, we should probably come up with a different name for this option (also it's _technically_ not really a preview because there is no "full" view of an operation which could be pre-viewed). Thoughts on naming it `overview` or `info`?